bulk-ingester/src/test/java/org/alfresco/hxi_connector/bulk_ingester/processor/mapper/AlfrescoPropertyMapperTest.java

```diff
@@ -0,0 +1,489 @@
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String description = "description";
+        String usDescription = "French fries recipe";
+        String ukDescription = "Chips recipe";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty("title"),
+                        createNodeProperty(description, "en_US_", stringValue(usDescription)),
+                        createNodeProperty(description, "en_UK_", stringValue(ukDescription))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(description, property.get().getKey());
+        assertThat(List.of(usDescription, ukDescription)).hasSameElementsAs((Iterable<String>) property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessStringProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String description = "description";
+        String descriptionText = "This document is about different animals legs length";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(description, stringValue(descriptionText))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(description, property.get().getKey());
+        assertEquals(descriptionText, property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessNullProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String description = "description";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(description, nullValue())));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, description).performMapping();
+
+        // then
+        assertFalse(property.isPresent());
+    }
+
+    @Test
+    void shouldProcessBoolProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String isIndexed = "isIndexed";
+
+        alfrescoNode.setNodeProperties(
+                Set.of(
+                        createNodeProperty(isIndexed, boolValue(true))));
+
+        // when
+        var property = new AlfrescoPropertyMapper(alfrescoNode, isIndexed).performMapping();
+
+        // then
+        assertTrue(property.isPresent());
+        assertEquals(isIndexed, property.get().getKey());
+        assertTrue((Boolean) property.get().getValue());
+    }
+
+    @Test
+    void shouldProcessLongProperties()
+    {
+        // given
+        AlfrescoNode alfrescoNode = new AlfrescoNode();
+
+        String legsCount = "legsCount";
+        long legsCountValue = 3;
```

Comment:
Using long for a count of the number of legs :scream:

Reply:
in case someone would like to put here number of legs of `centipede` from "James and the Giant Peach" ;)